src/UIModule.php

```diff
@@ -0,0 +1,202 @@
+            $this->_getConfig()['metabox']['title'],
+            function () use ($c) {
+                return $this->_renderMetabox($c);
+            },
+            $this->_getConfig()['metabox']['post_type']
+        );
+
+        /*
+         * Add screen options on bookings management page.
+         */
+        $eventManager->attach('screen_settings', function($settings, $screen = null) use ($c) {
+            if ($this->bookingsPageId !== $screen->base)
+                return $settings;
+
+            return $this->_renderBookingsScreenOptions($c);
+        });
+    }
+
+    /**
+     * Register pages in the admin menu.
+     *
+     * @param ContainerInterface $c
+     */
+    protected function _adminMenu(ContainerInterface $c)
+    {
+        $this->bookingsPageId = add_menu_page(
+            $this->_getConfig()['menu']['root']['page_title'],
+            $this->_getConfig()['menu']['root']['menu_title'],
+            $this->_getConfig()['menu']['root']['capability'],
+            $this->_getConfig()['menu']['root']['menu_slug'],
+            function () use ($c) {
+                return $this->_renderMainPage($c);
+            },
+            $this->_getConfig()['menu']['root']['icon'],
+            $this->_getConfig()['menu']['root']['position']
+        );
+
+        add_submenu_page(
+            $this->_getConfig()['menu']['root']['menu_slug'],
+            $this->_getConfig()['menu']['settings']['page_title'],
+            $this->_getConfig()['menu']['settings']['menu_title'],
+            $this->_getConfig()['menu']['settings']['capability'],
+            $this->_getConfig()['menu']['settings']['menu_slug'],
+            function () use ($c) {
+                return $this->_renderSettingsPage($c);
+            }
+        );
+
+        add_submenu_page(
+            $this->_getConfig()['menu']['root']['menu_slug'],
+            $this->_getConfig()['menu']['about']['page_title'],
+            $this->_getConfig()['menu']['about']['menu_title'],
+            $this->_getConfig()['menu']['about']['capability'],
+            $this->_getConfig()['menu']['about']['menu_slug'],
+            function () use ($c) {
+                return $this->_renderAboutPage($c);
+            }
+        );
+    }
+
+    /**
+     * Render screen options
+     *
+     * @param ContainerInterface $c
+     * @return mixed
+     * @throws \Psr\Container\ContainerExceptionInterface
+     * @throws \Psr\Container\NotFoundExceptionInterface
+     */
+    protected function _renderBookingsScreenOptions(ContainerInterface $c)
+    {
+        return $c->get('template-manager')->render('booking/screen-options');
+    }
+
+    /**
+     * Register metabox for service's bookings settings.
+     *
+     * @param ContainerInterface $c
+     * @return string
+     *
+     * @throws \Psr\Container\ContainerExceptionInterface
+     * @throws \Psr\Container\NotFoundExceptionInterface
+     */
+    protected function _renderMetabox($c)

```

Comment:
I think it makes more sense for this method, as well as the other similar methods (`_renderMainPage()`, `_renderSettingsPage()`, `_renderAboutPage`) to accept a `TemplateManager` instance instead of the container. The less the container is passed around, the better. I would only make exceptions if a class/method would otherwise need 5 or more parameters.